model/model.go

```diff
@@ -764,13 +768,39 @@ func (pi *PartitionInfo) GetNameByID(id int64) string {
 	return ""
 }
 
+func (pi *PartitionInfo) GetStateByID(id int64) (SchemaState, bool) {
+	for _, pstate := range pi.Definitions {
+		if pstate.ID == id {
+			return pstate.State, true
+		}
+	}
+	return StateNone, false
+}
+
+func (pi *PartitionInfo) SetStateByID(id int64, state SchemaState) bool {
+	for i, pstate := range pi.Definitions {
+		if pstate.ID == id {
+			pstate.State = state
+			pi.Definitions[i] = pstate
+			return true
+		}
+	}
+	return false
+}
+
+type PartitionState struct {
+	ID    int64       `json:"id"`
+	State SchemaState `json:"state"`
+}
+
 // PartitionDefinition defines a single partition.
 type PartitionDefinition struct {
-	ID       int64      `json:"id"`
-	Name     CIStr      `json:"name"`
-	LessThan []string   `json:"less_than"`
-	InValues [][]string `json:"in_values"`
-	Comment  string     `json:"comment,omitempty"`
+	ID       int64       `json:"id"`
+	Name     CIStr       `json:"name"`
+	LessThan []string    `json:"less_than"`
+	InValues [][]string  `json:"in_values"`
+	State    SchemaState `json:"state"`
```

Comment:
There is a compatibility problem here. After upgrading TiDB from an older version, since there was no field `State` on TiKV, so the `State` field is filled with 0 value, which corresponds to `StateNone`. However, `StateNone` means neither readable nor writable.

Reply:
yes @tiancaiamao @bb7133  PTAL

Reply:
use `PartitonState` instead.